src/Microsoft.Diagnostics.Monitoring.EventPipe/Triggers/AspNet/AspNetRequestStatusTriggerSettings.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe.Triggers.AspNet
+{
+    internal sealed class AspNetRequestStatusTriggerSettings : AspNetTriggerSettings
+    {
+        private const string StatusRegex = "[1-5][0-9]{2}";
+        private static readonly string RangeRegex = FormattableString.Invariant($"^{StatusRegex}(-{StatusRegex})?$");
+
+        /// <summary>
+        /// Specifies the set of status codes for the trigger. This can be individual codes or ranges.
+        /// E.g. 200;400-500
+        /// </summary>
+        [Required]
+        public string[] StatusCodes { get; set; }
+
+        public override IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
```

Comment:
FYI, IValidateObject.Validate is only run if all of the attribute rules pass. So if some other Required or Range rule fails from the base class, then this will not get run. Consider making this a ValidationAttribute implementation so that it runs inline with the other attributes that way all validation errors are provided at once. If you want to get fancy, you might be able to create something like a ItemsValidationAttribute that can apply other ValidationAttribute implementations to each individual item; this might be tricky because the items are strings as opposed to some custom class that you can put the attributes onto.

Maybe take this as a separate work item since there is technically nothing wrong with this implementation.